server/core/src/main/scala/org.apache.daffodil.debugger.dap/DAPodil.scala

```diff
@@ -358,11 +363,11 @@ class DAPodil(
           _ <- data.stack
             .findFrame(DAPodil.Frame.Id(args.frameId))
             .fold(
-              Logger[IO].warn(
+              Logger[IO].error(
```

Comment:
Can you introduce a one-liner method call to replace all these 5 lines of error-checking and handling? 

This looks like an abend to me. You are logging the error. So it appears this is never supposed to happen, i.e., one is technically always supposed to find scopes for the frames.